tests/tensor/rewriting/test_math.py

```diff
@@ -2010,6 +2010,27 @@ def test_exp_softplus(self, exp_op):
             decimal=6,
         )
 
+    def test_softplus_log(self):
+        # softplus(log(x)) -> log1p(x)
+        data_valid = np.random.random((4, 3)).astype("float32") * 2
+        data_valid[0, 0] = 0  # edge case
+        data_invalid = data_valid - 2
+
+        x = fmatrix()
+        f = function([x], softplus(log(x)), mode=self.mode)
```

Comment:
if you want you can check against the expected graph directly, something like 

```python
assert equal_computations(f.maker.fgraph.outputs, [pt.switch(x > 0, pt.log1p(x), np.asarray([[np.nan]], dtype="float32")])
```

Or something like that. This is not a request!

Reply:
Yeah nvm. I usually use it with just the output of `rewrite_graph` with a subset of rewrites. Not worth the trouble here